### PR TITLE
Update ppxlib to use cinaps clause and helpers library.

### PR DESCRIPTION
**Pull request by ceastlund**
Upstreaming this from code written at janestreet (adapted for dune rather than jenga).

### CODE REVIEW

File: src/cinaps/dune [hunk 1]
```diff
@@ -0,0 +1,4 @@
+(library
+ (name ppxlib_cinaps_helpers)
+ (public_name ppxlib.cinaps_helpers)
```
Comment (diml) [inline src/cinaps/dune]: Why is this library public?

Comment (ceastlund) [reply src/cinaps/dune]: Hah! Sorry, I was just used to writing dune files that way.

**Comment by diml** [comment]
I just submitted a new release of cinaps with the staged mode: https://github.com/ocaml/opam-repository/pull/15015. Once it's accepted we can restart Travis and the build should succeed

**Comment by ceastlund** [comment]
Weird, it worked for some versions of ocaml and not for others. @diml, do you have any insight into the errors?

**Comment by diml** [comment]
The cinaps dependency is missing for tests. I tried adding it. I don't know why it succeed at all on some of the versions :confused:

**Comment by diml** [comment]
I restarted the build now that the new cinaps has been released. Let see how it goes. Could you sign your commits BTW?

**Comment by ceastlund** [comment]
Re-pushed with signed-off commits.

**Comment by diml** [comment]
Good to merge as soon as CI passes

**Comment by diml** [comment]
I added you to the repository so you should be able to merge this PR

**Comment by ceastlund** [comment]
Still failing. Looks like some package constraint problems. Is this just due to the cinaps pin we've been doing?

**Comment by diml** [comment]
I found the issue: https://github.com/ocaml/opam-repository/pull/15209

**Comment by ceastlund** [comment]
Do we need to wait for an opam release to pick up the fix?

**Comment by diml** [comment]
Yh, the opam PR has been merged. I restarted Travis, hopefully it should succeed this time!

**Comment by diml** [comment]
And we can ignore the AppVeyor failures for now, they seem unrelated

**Comment by diml** [comment]
Ok, now Travis is happy except for older OCaml versions. Indeed, cinaps doesn't build with OCaml < 4.07...

**Comment by diml** [comment]
hmm, this is kind of annoying. I feel like the simplest choices are:

1. drop compat for OCaml < 4.07 in ppxlib
2. adapt cinaps so that it works with OCaml >= 4.04

1 obviously requires much less work than 2. What do others think? /cc @trefis @xclerc

**Comment by ceastlund** [comment]
I mean a lot of the new ppx compatibility work is aimed at making ppx compatible over many versions, so it seems worth some effort to keep 4.04+ compatibility, doesn't it?

**Comment by diml** [comment]
Yes, well after you can also say that if you use an older compiler then you use an older ppxlib. We just need to keep a window of reasonable size. That'll be the same for ppx, initially we should support a wide range of version but eventually will start dropping support for older versions of the compiler.

**Comment by ceastlund** [comment]
I do not appear to have write access, so I can't merge this.

**Comment by diml** [comment]
github says that you haven't accepted the invitation to join the project yet.

Just had a chat with @trefis and @xclerc, the easiest way to keep the CI green without more work is to only do the build with older compilers in Travis and build&run the tests only with newer compilers.

**Comment by diml** [comment]
Travis is happy now. Merging

**Pull request merged**